File: lib/tyto/entities/assignment.rb
module Tyto
  class Assignment < Entity
    attr_reader :id,
                :student_id,
                :chapter_id,
                :classroom_id,
                :complete,
                :course_name,
                :assignment_size,
                :teacher_id,
                :name,
                :questions_answered,
                :current_streak,
                :longest_streak,
                :current_question_text,
                :proficiencies,
                :proficiency,
                :question_level,
                :deadline,
                :subname,
                :video_url
    validates_presence_of :id, :student_id, :chapter_id, :classroom_id, :assignment_size, :teacher_id
  end
end

File: web/app/assets/javascripts/assignments/model/assignment.js
(function() {
  window.Assignment = function (assignment){
    this.id = assignment.id;
    this.questionText = assignment.current_question_text;
    this.questionsAnswered = assignment.questions_answered;
    this.assignmentSize = assignment.assignment_size;
    this.currentStreak = assignment.current_streak;
    this.longestStreak = assignment.longest_streak;
    this.proficiency = assignment.proficiency;
    this.questionLevel = assignment.question_level;
    this.proficiencies = assignment.proficiencies;
    this.complete = assignment.complete;
  };
})();

